src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -347,6 +352,11 @@ public void MoveTo(String destinationEntryName)
 
         #region Privates
 
+        // Only allow opening ZipArchives with large ZipArchiveEntries in update mode when running in a 64-bit process.
+        // This is for compatibility with old behavior that threw an exception for all process bitnesses, because this
+        // will not work in a 32-bit process.
```

Comment:
I don't think it will ever be successful in a 32-bit application because of the lack of available virtual address space.  That leaves me in the position of deciding between being compatible between 64-bit OOM and 32-bit OOM in corefx, or compatible between corefx and Desktop.  I think it will have little practical impact either way, but between those choices I think it's marginally better to be compatible with Desktop.